### PR TITLE
feat: add articles audio player

### PR DESCRIPTION
**Pull request by shahin-hq**
## Summary
Closes: https://app.clickup.com/t/8678v0juv

This  PR implements two audio players:
- One with use of a [library - wavesurfer](https://github.com/katspaugh/wavesurfer.js) and achieves the exact design
- Another is completely custom but has a different progress bar

image

image

**For testing:**
- Run `ArticleSeeder`
- Set this URL as `audio_file_url`
```
https://dashbrd-testing.s3.eu-west-1.amazonaws.com/24/en.0199216d-8e1d-419c-b207-e8c0a611f969.mp3
```

Try both and pick your side

image

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

**Review by crnkovic — COMMENTED** [review]
I personally like the wave surfer more, looks sleek! Will see what others say!

I just noticed the tiny color issue for the waves (should be primary-900) and forgotten translations in "Audio Version".

Screenshot 2023-10-27 at 13 56 50

File: resources/icons/audio-play.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg width="20" height="21" viewBox="0 0 20 21" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M12.852 6.67c2.724 1.676 4.085 2.514 4.341 3.68.087.398.087.809 0 1.206-.256 1.166-1.617 2.004-4.34 3.68C9.89 17.06 8.41 17.97 7.21 17.62a2.812 2.812 0 0 1-1.095-.611c-.928-.838-.928-2.577-.928-6.055 0-3.478 0-5.217.928-6.054a2.813 2.813 0 0 1 1.095-.612c1.2-.351 2.68.56 5.642 2.383Z" fill="currentColor"/></svg>
```
Comment (ItsANameToo) [inline resources/icons/audio-play.svg:1]: best to remove width and height and only keep the viewBox so it's possible to use different sizes for the image. Same applies to the other one

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
I prefer the wavesurfer one as well, looks good. 

One remark about the loading state for it: the spinner takes up more height than the actual wave it will show and it shifts the content

**Review by ItsANameToo — APPROVED** [review]

File: resources/js/Pages/Articles/Components/AudioPlayer.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        const rect = progressBarReference.current.getBoundingClientRect();
+        const x = "touches" in event ? event.touches[0].clientX : event.clientX;
+        let percent = ((x - rect.left) / rect.width) * 100;
+
+        percent = percent > 100 ? 100 : percent;
+        percent = percent < 0 ? 0 : percent;
+
+        const newTime = (percent / 100) * duration;
+        audioElement.currentTime = newTime;
+        setCurrentTime(newTime);
```
[the remark this answers is not in the document]
Comment (shahin-hq) [reply resources/js/Pages/Articles/Components/AudioPlayer.tsx]: Removed :sadup:

**Pull request merged**